File: day03/part1.nim
import nre, options, strutils, sequtils, parseutils
import intsets

var ids = initIntSet()
var grid = newSeqWith(1000, newSeq[int](1000))
var total = 0
# claims = []
for line in "input".lines():
  let f = line.find(re"""#(\d+) \@ (\d+),(\d+): (\d+)x(\d+)""")
  var claim = map(f.get.captures.toSeq(), proc(x: string): int = parseInt(x))
  var left = claim[1]
  var top = claim[2]
  var width = claim[3]
  var height = claim[4]
  
  for row in top .. (top + height - 1):
    for col in left .. (left + width - 1):
      grid[row][col] += 1
      if grid[row][col] == 2:
        total += 1

echo total

File: day03/part1.py
import re


class Grid(object):
    def __init__(self, size=1000):
        self.grid = [[0 for i in range(size)] for u in range(size)]

    def add(self, left, top, width, height):
        for row in range(top, top+height):
            for col in range(left, left+width):
                self.grid[row][col] += 1

    def overlap(self):
        t = 0
        for row in self.grid:
            for val in row:
                if val >= 2:
                    t += 1
        return t


grid = Grid()
claims = []
r = re.compile(r"""#(\d+) \@ (\d+),(\d+): (\d+)x(\d+)""")
for line in open("input", "r").readlines():
    f = r.findall(line.strip())
    claim = map(int, f[0])
    i, left, top, width, height = claim
    claims.append((left, top, width, height))
    grid.add(left, top, width, height)
print(grid.overlap())
